### PR TITLE
Fix CLI for latest Werkzeug (2.0.0rc1)

### LINKED ISSUE #3909: CLI incompatibility with Werkzeug 2 LocalProxy
LocalProxy has been changed in Werkzeug 2:

> LocalProxy matches the current Python data model special methods, including all r-ops, in-place ops, and async. __class__ is proxied, so the proxy will look like the object in more cases, including isinstance. Use issubclass(type(obj), LocalProxy) to check if an object is actually a proxy. :issue:`1754`

The Flask CLI can fail to run if the following conditions are met:

1. A `LocalProxy` is defined that requires an app context.
2. `FLASK_APP` is not defined.
3. The app is not named `app` or `application` or an app factory is used.

A simple `app.py` to reproduce:
```
from flask import current_app, Flask
from werkzeug.local import LocalProxy

log = LocalProxy(lambda: current_app.logger)

def create_app():
    app = Flask(__name__)
    return app
```

Then, running `flask shell` should drop nicely into the console. Instead, the result is `RuntimeError: Working outside of application context.`

Environment:

- Python version: 3.9
- Flask version: 1.1.2
- **Werkzeug version: 2.0.0rc1**

The issue is with this line: 
https://github.com/pallets/flask/blob/adeaf27e767f7bd37996d999eba135d63e890408/src/flask/cli.py#L51

and the aforementioned "the proxy will look like the object in more cases, including isinstance."

We can wrap the `isinstance` check in a try/except to ignore failing objects.

### PR DESCRIPTION
**Pull request by ziddey**
Wrap the `isinstance` check in a try/except to ignore failing objects (LocalProxy outside app context).

"LocalProxy matches the current Python data model special methods, including all r-ops, in-place ops, and async. class is proxied, so the proxy will look like the object in more cases, including isinstance. Use issubclass(type(obj), LocalProxy) to check if an object is actually a proxy. :issue:1754"

- fixes #3909 

Checklist:

- [X] Add tests that demonstrate the correct behavior of the change. Tests should fail without the change.
- [ ] Add or update relevant docs, in the docs folder and in code.
- [ ] Add an entry in `CHANGES.rst` summarizing the change and linking to the issue.
- [ ] Add `.. versionchanged::` entries in any relevant code docs.
- [X] Run `pre-commit` hooks and fix any issues.
- [X] Run `pytest` and `tox`, no tests failed.

### CODE REVIEW

**Comment by davidism** [comment]
I think we came up with a better solution to this by telling `LocalProxy` about its type even when unbound. I'll leave this open until I make sure that works.

File: src/flask/cli.py [hunk 1]
```diff
@@ -48,7 +48,13 @@ def find_best_app(script_info, module):
...
 
     # Otherwise find the only object that is a Flask instance.
-    matches = [v for v in module.__dict__.values() if isinstance(v, Flask)]
+    matches = []
+    for v in module.__dict__.values():
+        try:
+            if isinstance(v, Flask):
+                matches.append(v)
+        except Exception:
+            pass
```
Comment (recepsirin) [inline src/flask/cli.py:57]: Please, handle the possible exceptions because unless you handle them both explicitly or implicitly are meaningless.

**Review by recepsirin — CHANGES_REQUESTED** [review]
Please, handle the possible exceptions because unless you handle them both explicitly or implicitly are meaningless.

**Comment by davidism** [comment]
Fixed in https://github.com/pallets/werkzeug/pull/2042, specifically 352cd7167984f805daf226966f38f8dfeb991313.

**Pull request closed without merging**